src/QsCompiler/Transformations/LiftLambdas.cs

```diff
@@ -116,97 +117,58 @@ public override TypedExpression OnTypedExpression(TypedExpression ex)
                     }
                 }
 
-                private ParameterTuple MakeLambdaParams(ResolvedType expressionType, QsSymbol paramNames)
+                private ParameterTuple MakeLambdaParams(ParameterTuple paramNames)
                 {
                     var missingSymbolCount = 0;
 
-                    ParameterTuple MatchNameWithType(ResolvedType paramType, QsSymbol paramName)
+                    ParameterTuple MatchNameWithType(ParameterTuple paramName)
                     {
-                        if (paramName.Symbol.IsMissingSymbol)
+                        if (paramName is ParameterTuple.QsTupleItem param)
                         {
-                            var localVar = new LocalVariableDeclaration<QsLocalSymbol>(
-                                QsLocalSymbol.NewValidName($"__missingLambdaParam_{missingSymbolCount}__"),
-                                paramType,
-                                new InferredExpressionInformation(false, false),
-                                QsNullable<Position>.Null,
-                                paramName.Range.IsNull
-                                    ? DataTypes.Range.Zero
-                                    : paramName.Range.Item);
-                            missingSymbolCount++;
-                            return ParameterTuple.NewQsTupleItem(localVar);
+                            return param.Item.VariableName.IsInvalidName
```

Comment:
How equivalent is an invalid variable name with the missing symbol case? Will this always catch the missing symbols `_`? Will it catch more cases than it did before?

Reply:
We currently don't support underscores as parameter names (i.e. missing parameter names) in global callable declarations. This is something that may be nice to support in the future, but felt out of scope for this change. At the same time, it is nice to support it in particular for lambdas, since there the argument also doesn't require a type annotation. 
To align with how argument tuples work in general and still permit _ for parameters in lambdas, I needed to represent missing symbols as invalid variables. So the answer to your question is that missing symbols will indeed always be invalid variables, and - unless the compilation generated errors before, an invalid symbol here is indeed a missing symbol (`_`) in the source code. However, for invalid code such as e.g. `(a, a) -> a`), the second `a` here would also be represented as invalid variable name and would be replaced with an autogenerated name by this transformation. I think that is fine.